### PR TITLE
haproxy_auto_https: Ensure it works over HTTP even if ACME fails

### PR DESCRIPTION
**Pull request by aegamesi**
Previously, if the initial certificate could not be fetched (improper
configuration, Let's Encrypt is temporarily down, etc.), HAProxy would
never be started. This change ensures there are two copies of the
HAProxy configuration file (one with HTTPS and one without) to ensure
HAProxy can always be started with a valid configuration, even
without a certificate present.

### CODE REVIEW

File: docker/glue.py [hunk 1]
```diff
@@ -17,13 +17,20 @@ def start_haproxy(prev_pid=None):
...
     """
     cmd = ['haproxy', '-D']
     if prev_pid is not None:
         cmd += ['-sf', str(prev_pid)]
-    cmd += ['--', '/usr/local/etc/haproxy/haproxy.cfg']
+    cmd += ['--']
+    if os.path.isfile(path_combined):
+        cmd += ['/usr/local/etc/haproxy/haproxy.cfg']
+    else:
+        cmd += ['/usr/local/etc/haproxy/haproxy-no-https.cfg']
```
Comment (luise) [inline docker/glue.py]: I'd probably print some noticeable warning saying that the proxy is not running with HTTPS in this case.

Comment (luise) [reply docker/glue.py]: Seems like the proxy now always starts without HTTPS and then might later be restarted with HTTPS. If so, I guess there should also be a message when the proxy starts with HTTPS.

On a related note, it seems a little strange to me to always start _without_ HTTPS -- did I misunderstand?

**Comment by kklin** [comment]
I think this behavior is a bit dangerous. Is it really necessary?

If you're running an application that _needs_ HTTPS to protect user info (passwords, credit card info, etc), it'd be better to not start at all than to start insecurely IMO.

**Comment by aegamesi** [comment]
Yeah... perhaps it would be good to make it an option. I'm only concerned because of the lack of persistent storage -- if your container has to reboot or something and you can't refetch a certificate (maybe Let's Encrypt is temporarily down, or you've reached the rate limit somehow), your website will be inaccessible until you can get another certificate.

**Comment by kklin** [comment]
@aegamesi can we close this? I still think the behavior is a bit weird and possibly confusing, and there's no one asking for it anyways

**Pull request closed without merging**